todo-app/src/components/TasksForm.js

```diff
@@ -0,0 +1,30 @@
+import React, { Component } from 'react'
+import PropTypes  from 'prop-types'
+import TaskItem from './TaskItem'
+
+export default class TasksFrom extends Component {
+    render() {
+        const todos = this.props.todos
+        return (
+            <div>
+                {todos.map((task,index) => {
+                    return <TaskItem 
+                        key={index}
```

Comment:
The problem with deleting tasks is in this line @nos111! You are using the index as key, and as we mentioned during the last class, using the index might cause unexpected behaviour. React doesn't know that the item that removed with index 1 was different than the new item that is rendered (after deletion) at index 1.

What we want is to use the `task.id` as key and deleting items should work as expected. If you use chrome browser and the react devtools extension, then this is what we should se

<img width="429" alt="screen shot 2017-07-24 at 20 05 11" src="https://user-images.githubusercontent.com/1057324/28537610-cd00f046-70ab-11e7-890f-c463433c0325.png">

Notice the keys are now 1, 2, 3, 4 and not 0, 1, 2, 3 when you use the index as key 😄